src/Model/Model.ts

```diff
@@ -682,7 +682,7 @@ export class Model<T extends IModelData> implements IModel<T> {
    *
    * @returns {boolean}
    */
-  public isFieldDirty(fieldName) {
+  public isFieldDirty(fieldName: keyof T) {
```

Comment:
add some advanced type checking here, so TS will bark at you if you try and call it with a property that isnt on the model.